docs/cli-mode.md

```diff
@@ -0,0 +1,281 @@
+# CLI Mode
+The CLI mode can be used for bulk resource management. 
+
+Usages:
+* Export all/selected resources from an Identity Server to a local directory.
+* Import all/selected resources from a local directory to an Identity Server.
+* Propagate resources across multiple environments.
+* Deploy new resources from resource configuration files to an Identity Server.
+* Have a backup of resources in a local directory.
+
+Currently, supported resource types are:
+* Applications
+* Identity Providers
+
+## Running the Tool in the CLI Mode
+### Initialization
+Before running the tool to export or import any resource, the tool should be configured against the target environment.
+
+First, create the config files using the ```setupCLI``` command:
+```
+iamctl setupCLI
+```
+Use the ```--help``` flag to get more information on the command.
+   ```
+   Flags:
+    -d, --baseDir string   Path to the base directory
+    -h, --help             help for setupCLI
+   ```
+The above command will create a new folder named ```configs``` which will contain all the config files needed to setup the tool. 
+The folder structure of the ```configs``` directory will be as follows:
+```
+configs
+└── env
+    │── serverConfig.json
+    └── toolConfig.json
+``` 
+   It is recommended to place the ```configs``` folder inside the local directory created to maintain the resource configuration files. 
+   
+Example local directory structure if multiple environments: dev, stage, prod exist:
+   ```
+   local directory
+   │── configs
+   │    │── dev
+   │    │    │── serverConfig.json
+   │    │    └── toolConfig.json
+   │    │── stage
+   │    │    │── serverConfig.json
+   │    │    └── toolConfig.json
+   │    └──── prod
+   │         │── serverConfig.json
+   │         └── toolConfig.jsondev
+   │── Applications
+   │    │── app1.yml
+   │    │── app2.yml
+   │    │── ... other exported app files
+   │
+   │── IdentityProviders
+   │    │── idp1.yml
+   │    │── idp2.yml
+   │    │── ... other exported idp files
+   │
+   │── ... other resource types
+   ```
+   Use the ```--baseDir``` flag to specify the path to the local directory when creating the ```configs``` folder. If not specified, the tool will create the ```configs``` folder in the current directory.
+
+### Server Configurations
+Server configurations are the configurations needed to connect to the target environment. The server configurations can be provided through the ```serverConfig.json``` file or through environment variables. It is mandatory to provide the following parameters relevant to the target Identity Server to run the CLI commands.
+* Server URL of the target IS
```

Comment:
```suggestion
* Server URL of the target identity server
```

Reply:
I am confused about whether `IS` is `WSO2 IS` or any identity server. If it the former, use the term `identity server`. Otherwise, use `WSO2 IS`.